src/main/scala/org/apache/spark/metrics/sink/JdbcSink.scala

```diff
@@ -68,13 +70,25 @@ class JdbcSink(val properties: Properties, val registry: MetricRegistry,
   user.foreach(dataSource.setUsername)
   password.foreach(dataSource.setPassword)
 
+  val counterCache: concurrent.Map[String, Long] = concurrent.TrieMap.empty[String, Long]
   val reporter: JdbcReporter = {
     val builder = JdbcReporter.forRegistry(registry)
       .convertDurationsTo(TimeUnit.MILLISECONDS)
       .convertRatesTo(TimeUnit.SECONDS)
     for (f <- filter) {
       builder.filter(new MetricFilter {
-        override def matches(name: String, metric: Metric): Boolean = name.contains(f)
+        override def matches(name: String, metric: Metric): Boolean = {
+          lazy val hasChanged = metric match {
```

Comment:
Probably no point in making this lazy if we always evaluate it?

Reply:
Now we only evaluate it (or I hope scala is smart enough) when `name.contains(f)`